### ISSUE TITLE
PDF structure issue, 50 page limit

### ISSUE THREAD

**Issue by wangp** [nvaccess/nvda#7730]
I am testing NVDA 2017.3 (installed) with Acrobat Pro DC on Windows 7.

Compare these two files:

- [nvda-0049.pdf](https://github.com/nvaccess/nvda/files/1444586/nvda-0049.pdf) has 49 pages with one H1 tag per page. Pressing 'h' or '1' correctly skips to each heading.

- [nvda-0050.pdf](https://github.com/nvaccess/nvda/files/1444588/nvda-0050.pdf) has 50 pages with one H1 tag per page. Pressing 'h' or '1', NVDA reports that there is "no next heading". Furthermore, I cannot scroll through the document using the arrow keys.

In this second pair of test files, I have removed the page breaks before each H1, so that multiple headings appear on each page.

- [nvda-0441-no-break.pdf](https://github.com/nvaccess/nvda/files/1444597/nvda-0441-no-break.pdf) has 49 pages. Pressing 'h' or '1' correctly skips to each heading.

- [nvda-0442-no-break.pdf](https://github.com/nvaccess/nvda/files/1444598/nvda-0442-no-break.pdf) has 50 pages (1 extra page). Pressing 'h' or '1' only allows me to skip to the last heading on the first page. After that, NVDA reports "no next heading" and the arrow keys will not take me past the first page.

So it seems like there is an internal limit of 50 pages somewhere. As far as I can tell, the PDFs are syntactically correct and acceptably tagged.

**Comment by ehollig** [comment]
In Adobe Acrobat Pro DC, press CTRL+K to go to preferences. Press tab three times to go to "Page vs Document:  combo box". NVDA reads, "For large documents, only read the currently visible pages". Press tab one more time and notice the visible pages is 50.
Does this help your issue?

**Comment by wangp** [comment]
Indeed it does, thank you!

After I set "Page vs Document" to "Read the entire document", I was able to navigate within a few other large documents, after waiting for them to load in the background.
